packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -106,87 +163,200 @@ export class BlobManager {
         return `/${BlobManager.basePath}/${blobId}`;
     }
 
-    public async getBlob(blobId: string): Promise<IFluidHandle<ArrayBufferLike>> {
-        const storageId = this.redirectTable?.get(blobId) ?? blobId;
-        assert(this.hasBlob(storageId), 0x11f /* "requesting unknown blobs" */);
+    private get storageIds(): Set<string> {
+        const ids = new Set<string | undefined>(this.redirectTable.values());
+
+        // If we are detached, we will not have storage IDs, only undefined
+        const havePendingBlobs = ids.delete(undefined);
+
+        // For a detached container, entries are inserted into the redirect table with an undefined storage ID.
+        // For an attached container, entries are inserted w/storage ID after the BlobAttach op round-trips.
+        assert(!havePendingBlobs || this.runtime.attachState === AttachState.Detached && ids.size === 0,
+            "'redirectTable' must contain only undefined while detached / defined values while attached");
+
+        return ids as Set<string>;
+    }
+
+    public async getBlob(blobId: string): Promise<ArrayBufferLike> {
+        const pending = this.pendingBlobs.get(blobId);
+        if (pending) {
+            return pending.blob;
```

Comment:
I'm guessing that we intentionally do not want to call 'gcNodeUpdate()' since the blob is not yet externally reachable?

(Adding a comment would be helpful.)